gnomad_qc/v4/annotations/insilico_predictors.py

```diff
@@ -89,6 +100,94 @@ def _load_cadd_raw(cadd_tsv) -> hl.Table:
+    transcripts of different genes, the scores are equal.
+    REVEL scores were downloaded from:
+       https://www.google.com/url?q=https%3A%2F%2Frothsj06.dmz.hpc.mssm.edu%2Frevel-v1.3_all_chromosomes.zip&sa=D&sntz=1&usg=AOvVaw2DS2TWUYl__0vqijzzxp5M
+       size ~648M, ~82,100,677 variants
+    REVEL's Ensembl ID is not from Ensembl 105, so we filter to transcripts
+    that are in Ensembl 105. The Ensembl 105 ID file was downloaded from Ensembl
+    105 archive. It contains the following columns:
+       Transcript stable ID, Ensembl Canonical, MANE Select
+    This deprecates the `has_duplicate` field present in gnomAD v3.
+
+    :return: Hail Table with REVEL scores for GRCh38.
+    """
+    revel_csv = "gs://gnomad-insilico/revel/revel-v1.3_all_chromosomes_with_transcript_ids.csv.bgz"
+    ensembl_ids = "gs://gnomad-insilico/ensembl/ensembl105id.tsv.bgz"
+
+    ht = hl.import_table(
+        revel_csv,
+        delimiter=",",
+        min_partitions=1000,
+        types={"grch38_pos": hl.tstr, "REVEL": hl.tfloat64},
+    )
+
+    logger.info("Annotating REVEL table...")
+    ht = ht.drop("hg19_pos", "aaref", "aaalt")
+    # drop variants that have no position in GRCh38 when lifted over from GRCh37
+    ht = ht.filter(ht.grch38_pos.contains("."), keep=False)
+    ht = ht.transmute(chr="chr" + ht.chr)
+    ht = ht.annotate(
+        locus=hl.locus(ht.chr, hl.int(ht.grch38_pos), reference_genome="GRCh38"),
+        alleles=hl.array([ht.ref, ht.alt]),
+        Transcript_stable_ID=ht.Ensembl_transcriptid.strip().split(";"),
+    )
+    ht = ht.select("locus", "alleles", "REVEL", "Transcript_stable_ID")
+    ht = ht.explode("Transcript_stable_ID")
+    ht = ht.key_by("Transcript_stable_ID")
+    print(ht.count())
+
+    logger.info("Import and process the ensembl ID file...")
+    ensembl_ids = hl.import_table(
+        ensembl_ids, min_partitions=200, impute=True, key="Transcript_stable_ID"
+    )
+    ensembl_ids = ensembl_ids.select("Ensembl_Canonical", "MANE_Select")
+    logger.info("Annotating revel HT with canonical and MANE Select transcripts...")
+    ht = ht.annotate(**ensembl_ids[ht.Transcript_stable_ID])
+    print(ht.count())
+
+    logger.info(
+        "Annotating REVEL scores for MANE Select transcripts and canonical"
+        " transcripts..."
+    )
+    ht = ht.key_by("locus", "alleles")
+    ht = ht.annotate(
+        revel_mane=hl.or_missing(ht.MANE_Select != "", ht.REVEL),
+        revel_canonical=hl.or_missing(ht.Ensembl_Canonical == "1", ht.REVEL),
+        revel_noncanonical=hl.or_missing(ht.Ensembl_Canonical != "1", ht.REVEL),
+    )
+
+    logger.info("Taking max REVEL scores for MANE Select transcripts...")
+    ht = ht.checkpoint(
+        "gs://gnomad-tmp-4day/revel-v1.3_in_Ensembl105.ht", _read_if_exists=True
+    )
+    mane_ht = ht.filter(hl.is_defined(ht.revel_mane), keep=True)
+    max_revel_mane = mane_ht.group_by(*mane_ht.key).aggregate(
```

Comment:
Do you need the `*` in the group_by here? Isnt the key a single field?

Reply:
yes, it's keyed by locus and alleles now, not Transcript ID